src/Go/boardAnalysis/goAI.ts

```diff
@@ -137,6 +138,8 @@ export function resetAI(endOfGame = false): void {
     if (playerPromise.resolver) {
       playerPromise.resolver(gameOver);
       playerPromise.resolver = null;
+    } else {
+      playerPromise.nextTurn = Promise.resolve(gameOver);
     }
     if (!endOfGame && !playerPromise.resolver) {
       createPromise(playerPromise);
```

Comment:
```suggestion
    if (endOfGame) {
      playerPromise.nextTurn = Promise.resolve(gameOver);
    } else {
      createPromise(playerPromise);
```

I think what you had before would possibly fail in some subtle edge-cases involving multiple calls to resetAI. The real condition to be testing for is `endOfGame`. And, unless I'm missing something, the `!playerPromise.resolver` check is dead code (probably from an earlier iteration), so we can eliminate that and streamline this.

This breaks the idempotence of resetAI slightly, in that calling it repeatedly with `endOfGame = true` will continually create new promises. But, they all resolve to the same value, and generally it's bad form to be doing object identity checks on promise objects, and we don't call `endOfGame = true` multiple times anyway, really.

Reply:
I applied a slight variant of this (the proposed suggestion is missing a curly brace, I think)